### PR TITLE
Test return value for ENV.clear

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

**Comment by BurdetteLamar** [comment]
Is there something wrong here that prevents merging?  This just adds validation that the return value is ENV.

**Comment by eregon** [comment]
> Is there something wrong here that prevents merging? This just adds validation that the return value is ENV.

It requires some time for maintainers to review :wink:

File: core/env/clear_spec.rb [hunk 1]
```diff
@@ -3,8 +3,9 @@
 describe "ENV.clear" do
   it "deletes all environment variables" do
     orig = ENV.to_hash
+    env_object_id = ENV.object_id
     begin
-      ENV.clear
+      ENV.clear.object_id.should == env_object_id
```
Comment (eregon) [inline core/env/clear_spec.rb]: There is a matcher to do this more clearly (see https://github.com/ruby/spec/blob/master/CONTRIBUTING.md#matchers-and-expectations): 
```ruby
ENV.clear.should equal(ENV)
```
Could you use it?

Comment (BurdetteLamar) [reply core/env/clear_spec.rb]: Thanks, eregon.  I will update to use the matcher.  Also, I will study the matchers at the link.

**Pull request merged**